lib/datadog/open_feature/exposures/worker.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require_relative '../../core/utils/time'
+require_relative '../../core/workers/queue'
+require_relative '../../core/workers/polling'
+
+require_relative 'buffer'
+require_relative 'batch_builder'
+
+module Datadog
+  module OpenFeature
+    module Exposures
+      # This class is responsible for sending exposures to the Agent
+      class Worker
+        include Core::Workers::Queue
+        include Core::Workers::Polling
+
+        GRACEFUL_SHUTDOWN_EXTRA_SECONDS = 5
+        GRACEFUL_SHUTDOWN_WAIT_INTERVAL_SECONDS = 0.5
+
+        DEFAULT_FLUSH_INTERVAL_SECONDS = 30
+        DEFAULT_BUFFER_LIMIT = Buffer::DEFAULT_LIMIT
+
+        attr_reader :logger
+
+        def initialize(
+          settings:,
+          transport:,
+          logger:,
+          flush_interval_seconds: DEFAULT_FLUSH_INTERVAL_SECONDS,
+          buffer_limit: DEFAULT_BUFFER_LIMIT
+        )
+          @logger = logger
+          @transport = transport
+          @batch_builder = BatchBuilder.new(settings)
+          @buffer_limit = buffer_limit
+          @flush_mutex = Mutex.new
+
+          self.buffer = Buffer.new(buffer_limit)
+          self.fork_policy = Core::Workers::Async::Thread::FORK_POLICY_RESTART
+          self.loop_base_interval = flush_interval_seconds
+          self.enabled = true
+        end
+
+        def start
+          return if !enabled? || running?
+
+          perform
+        end
+
+        def stop(force_stop = false, timeout = Core::Workers::Polling::DEFAULT_SHUTDOWN_TIMEOUT)
+          buffer.close if running?
+
+          super
+        end
+
+        def enqueue(event)
+          start unless running?
+          buffer.push(event)
+
+          true
+        end
+
+        def dequeue
+          buffer.pop
+        end
+
+        def perform(*args)
+          events, dropped = args
+          send_events(Array(events), dropped.to_i)
```

Comment:
is `Array()` being called for nil-safeness? If so, I recommend checking for nil and returning early instead, since `Array()` will duplicate a potentially large `events` arrays in the first few calls with little value in return.